multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -71,20 +71,17 @@ func NewResourceExportReconciler(
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the ResourceExport object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *ResourceExportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	klog.V(2).InfoS("reconciling ResourceExport", "resourceexport", req.NamespacedName)
 	var resExport mcsv1alpha1.ResourceExport
 	if err := r.Client.Get(ctx, req.NamespacedName, &resExport); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !(resExport.Spec.Kind == common.ServiceKind || resExport.Spec.Kind == common.EndpointsKind) {
```

Comment:
I'm not sure if this is the best way to do this. For different kinds of resourceExports, some code down here can still be shared (pre-deletion logic for example). Should we simply put `refreshServiceResourceImport` etc. in separate files instead?

Reply:
yes, you are right, I was thinking to handle different kinds in different controllers, but considering there will introduce code duplication, I will refine it today. @jianjuns

Reply:
Hi @Dyanngg , I think the deletion logic might be different for different kinds, so I would leave it to you to decide to use common code or handle them in new file.